flink-core/src/main/java/org/apache/flink/configuration/MemorySize.java

```diff
@@ -158,6 +162,38 @@ private String formatToString() {
 			highestIntegerUnit.getUnits()[1]);
 	}
 
+	public String toHumanReadableString() {
+		if (humanReadableStr == null) {
+			humanReadableStr = formatToHumanReadableString();
+		}
+
+		return humanReadableStr;
+	}
+
+	private String formatToHumanReadableString() {
+		MemoryUnit highestUnit = IntStream.range(0, ORDERED_UNITS.size())
+			.sequential()
+			.filter(idx -> bytes > ORDERED_UNITS.get(idx).getMultiplier())
+			.boxed()
+			.max(Comparator.naturalOrder())
+			.map(idx -> ORDERED_UNITS.get(idx))
+			.orElse(BYTES);
+
+		if (highestUnit == BYTES) {
+			return String.format(
+				"%d %s",
+				bytes,
+				BYTES.getUnits()[1]);
+		} else {
+			double approximate = 1.0 * bytes / highestUnit.getMultiplier();
+			return String.format(
+				"%.3f%s (%d)",
```

Comment:
Should we add `%.3f%s (%d bytes)`?

Reply:
I thought people could easily understand that the value in the parentheses are they exact bytes. But true, it would be better to explicitly add the unit.